diff --git a/src/bench.c b/src/bench.c
index 399de1c..d645e1d 100644
--- a/src/bench.c
+++ b/src/bench.c
@@ -43,7 +43,8 @@
 
 #define cmp(a,b) (*(a) > *(b)) // uncomment for fast primitive comparisons
 
-char *sorts[] = { "*", "blitsort", "crumsort", "fluxsort", "gridsort", "quadsort", "wolfsort" };
+//char *sorts[] = { "*", "blitsort", "crumsort", "fluxsort", "gridsort", "quadsort", "wolfsort" };
+char *sorts[] = { "*", "quadsort", "pdqsort", "fluxsort", "wolfsort", "ska_sort", "rhsort" };
 
 #if __has_include("blitsort.h")
   #include "blitsort.h" // curl "https://raw.githubusercontent.com/scandum/blitsort/master/src/blitsort.{c,h}" -o "blitsort.#1"
